integration_tests/integration/serviceProviderReferrals.spec.js

```diff
@@ -1082,9 +1082,17 @@ describe('Service provider referrals dashboard', () => {
         currentAppointmentId: scheduledAppointment.id,
       })
       cy.stubScheduleSupplierAssessmentAppointment(supplierAssessment.id, scheduledAppointment)
-      cy.stubGetSupplierAssessment(referral.id, supplierAssessmentWithScheduledAppointment)
 
       cy.contains('Save and continue').click()
+      // We need to switch out the response _after_ the update, since the
```

Comment:
Ugh, I remember struggling with this a while ago and think I gave up and followed the same approach you have - good to know there's a solution for it though